plugins/modules/firewall.py

```diff
@@ -242,23 +242,57 @@ def _normalize_ips(rules: list) -> list:
 
         return result
 
-    def _rules_changed(self) -> bool:
-        """Returns whether the local and remote firewall rules match."""
+    def _amend_rules(self, remote_rules: list, local_rules: list) -> None:
```

Comment:
Small return type issue here
```suggestion
    def _amend_rules(self, remote_rules: list, local_rules: list) -> list:
```